cardano-db-sync/src/Cardano/DbSync/Fix/PlutusScripts.hs

```diff
@@ -109,18 +109,18 @@ findWrongPlutusScripts tracer =
       prevPoint <- convertToPoint (SlotNo prevSlotNo) prevBlockHsh
       Just prevPoint
 
+    hashPlutusScript :: DB_V_13_0.Script -> Either (Maybe String) ByteString
     hashPlutusScript dbScript = do
-      --      lang <- getLang -- TODO: Conway
-      bytes <- maybeToEither "No bytes found for plutus script" id $ DB_V_13_0.scriptBytes dbScript
-      let script :: AlonzoScript StandardAlonzo = PlutusScript (AlonzoPlutusV1 (Plutus $ PlutusBinary $ SBS.toShort bytes))
-      let hsh :: Ledger.ScriptHash StandardCrypto = Ledger.hashScript @StandardAlonzo script
-      Right $ Generic.unScriptHash hsh
-
---      where
---        getLang = case DB_V_13_0.scriptType dbScript of
---          PlutusV1 -> Right Ledger.PlutusV1
---          PlutusV2 -> Right Ledger.PlutusV2
---          _ -> Left "Non plutus script found where it shouldn't."
+      bytes <- maybeToEither (Just "No bytes found for plutus script") id $ DB_V_13_0.scriptBytes dbScript
+      case DB_V_13_0.scriptType dbScript of
+        PlutusV1 -> do
+          -- The bug only affected Alonzo script
+          let script :: AlonzoScript StandardAlonzo = PlutusScript (AlonzoPlutusV1 (Plutus $ PlutusBinary $ SBS.toShort bytes))
+          let hsh :: Ledger.ScriptHash StandardCrypto = Ledger.hashScript @StandardAlonzo script
+          Right $ Generic.unScriptHash hsh
+        PlutusV2 -> Left Nothing
+        PlutusV3 -> Left Nothing
+        _ -> Left $ Just "Non plutus script found where it shouldn't."
```

Comment:
not 100% clear on the message, does `A non-Plutus script was discovered in an unexpected location.` sound better?

Reply:
Leaving this as is for now. This comment existed before, but was commended out here https://github.com/IntersectMBO/cardano-db-sync/pull/1648/files/5f2881e04587c2a7cd25f695dfd5c4a3b453cc3a#diff-23041e6f296ec26493d184b4167eb2d3070f4483fcbfe1d5048bef429b209f3bL123